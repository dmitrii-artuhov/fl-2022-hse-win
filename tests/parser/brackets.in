start=🤯A🤯
🤯S🤯 👉 🤯lb🤯 🤯B🤯🗿
🤯A🤯 👉 🤯lb🤯 🤯B🤯 🤌 😵 🗿
🤯B🤯 👉 🤯S🤯  🤯C🤯 🤌 🤯rb🤯 🤯S🤯 🤌 🥵)🥵🗿
🤯C🤯 👉 🤯rb🤯 🤯S🤯 🤌 🥵)🥵🗿
🤯lb🤯 👉 🥵(🥵🗿
🤯rb🤯 👉 🥵)🥵🗿