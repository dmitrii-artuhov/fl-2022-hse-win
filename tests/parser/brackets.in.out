
-- Grammar --
Terminals: ['(', ')']
Non-terminals: ['A', 'B', 'C', 'S', 'lb', 'rb']
Start: 'A'
Rules:
'S' -> ['lb', 'B']
'A' -> ['lb', 'B'] | ['Îµ']
'B' -> ['S', 'C'] | ['rb', 'S'] | [')']
'C' -> ['rb', 'S'] | [')']
'lb' -> ['(']
'rb' -> [')']

Chomsky normal form: True
        
